### PR TITLE
chore: changelog dune 3.17.1

### PR DESCRIPTION
**Pull request by maiste**
This PR adds the changelog for the upcoming dune `3.17.1`.

### CODE REVIEW

**Comment by maiste** [comment]
@cuihtlauac @sabine the release being done, it is ready for review :smile:

File: data/changelog/dune/2024-12-18-dune.3.17.1.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    - Remove useless error message when running `$ dune subst` in empty projects.
+      (@rgrinberg, #11204, fixes #11200)
+---
+
+The Dune team is happy to announce the release of Dune 3.17.1!
+
+This patch release includes some bug fixes. To reduce computing time, it does
+not build `.cmxs` files anymore when the `(no_dynlink)` stanza is used instead.
+This behavior also corrects the semantic of the `(no_dynlink)` stanza which was
+building but not installing `.cmxs` files.
```
Comment (maiste) [inline data/changelog/dune/2024-12-18-dune.3.17.1.md]:
```suggestion
building but not installing `.cmxs` files. It does not try to build and install them anymore.
```

**Comment by cuihtlauac** [comment]
@maiste: can you rebase from a fresh main? It was broken this morning. It should be fixed now

**Comment by maiste** [comment]
Done :heavy_check_mark:

**Pull request merged**